### DISCUSSION TITLE
Progress information for Activity bar and Viewlets

### DISCUSSION THREAD

**Discussion by daniel-a126** [microsoft/vscode-discussions#1240] [Extension Development QnA] [answered]
We are working on an extension that has its own _Activity bar_ and four _viewlets_ inside it. In certain conditions, our view may take some time to load when the extension is activating.

There are two kinds of progress bars in the side bar:
1. Global, overall for the whole side bar
2. Individual for each viewlet

It seems that there is no way for us to control them (API missing), but also, the documentation on when they show and why is missing.

It seems that the the _global_ is not used whatsoever and the activation/loading state is actually shown for the _individual_ viewlets. Is that like this for historical reasons? Or is there a scenario when the global progress is used? Why is not part of the progress API?

Can somebody help filling this gap?

**Comment by isidorn** [comment]
I do not know the exact history here, and how the progress API evolved.
But the goal should be that VS Code will automatically show progress in the right sport based on running operation, and the progress API allows extensions to explicitly choose a location. Window being in status bar, and notification showing a notification.

@bpasero might want to provide more context

**Reply by bpasero** [reply, marked as answer]
To my knowledge there is:
* per-view progress (by identifier of the view)
* window progress (status bar)
* notification progress

I am not aware of anything else.

Ideally progress location is picked as close as possible to where something is long running.

**Reply by daniel-a126** [reply]
Thank you; That makes sense; Historical reasons :))
